Build failure: pcl

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

- `nix-shell -p pcl`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/323010340

### Relevant log output

```console
CMake Error at outofcore/CMakeLists.txt:79 (target_link_libraries):
  Target "pcl_outofcore" links to:

    Boost::system

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.



-- Generating done (1.0s)
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY


CMake Generate step failed.  Build files cannot be regenerated correctly.
```

### Additional context

The problem is caused by Boost 1.89.0, where the `system` component was removed. However, this component remained [pcl_find_boost.cmake](https://github.com/PointCloudLibrary/pcl/blob/pcl-1.15.1/cmake/pcl_find_boost.cmake) when 1.15.1 released, and was only fixed 2 days later ([commit](https://github.com/PointCloudLibrary/pcl/commit/99333442ac63971297b4cdd05fab9d2bd2ff57a4)).

Just needs this added:
```
  patches = [
    (fetchpatch {
      # https://github.com/PointCloudLibrary/pcl/pull/6330
      url = "https://github.com/PointCloudLibrary/pcl/commit/99333442ac63971297b4cdd05fab9d2bd2ff57a4.patch";
      hash = ""; # whatever the hash is
    })
  ];
```

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.13, NixOS, 26.05 (Yarara), 26.05pre953160.2fc6539b481e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.3`
 - channels(root): `"home-manager, nixos"`
 - nixpkgs: `/nix/store/74lrn4968lcj1hhf53lp08wai0dbjamx-nixos/nixos`

### Notify maintainers

@GaetanLepage 
@usertam
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/pc/pcl/package.nix
+++ b/pkgs/by-name/pc/pcl/package.nix
@@ -3,6 +3,7 @@
   stdenv,
   config,
   fetchFromGitHub,
+  fetchpatch,
 
   # nativeBuildInputs
   cmake,
@@ -41,6 +42,15 @@ stdenv.mkDerivation (finalAttrs: {
     tag = "pcl-${finalAttrs.version}";
     hash = "sha256-+KyaajJM0I5CAcr8AiOLC4TkGV3Gm73a0/X8LQWFZMI=";
   };
+
+  patches = [
+    (fetchpatch {
+      # see https://github.com/NixOS/nixpkgs/issues/485826 to be removed at next release after 1.15.1
+      name = "boost-1.89.patch";
+      url = "https://github.com/PointCloudLibrary/pcl/commit/99333442ac63971297b4cdd05fab9d2bd2ff57a4.patch";
+      hash = "sha256-5vg8VjxoAfEOx9n7Tby1DXe1u4rn+zharkefUovLHv0=";
+    })
+  ];
 
   strictDeps = true;
 

